Tests/RxSwiftTests/PrimitiveSequenceTest.swift

```diff
@@ -793,16 +793,28 @@ extension PrimitiveSequenceTest {
     }
 
     #if TRACE_RESOURCES
-        func testAsSingleReleasesResourcesOnComplete() {
+        func testAsSingleReleasesResourcesOnComplete1() {
             _ = Observable<Int>.just(1).asSingle().subscribe({ _ in })
         }
 
+        func testAsSingleReleasesResourcesOnComplete2() {
```

Comment:
`testAsSingleReleasesResourcesOnComplete1` is a test does observable sequence ( some particular operator) releases all of it's resources without `dispose` being called explicitly.

In this case `asSingle` operator is being tested.

I think we need to create unit tests that test is block called for a particular sequence.

```
func testSingle_subscribeOnSuccess() {
var events: [SingleEvent<Int>] = []

_ = Single.just(1).subscribe(onSuccess: { element in
     events.add(.success(element))
}, onError: { error in
     events.add(.error(error))
})

XCTAssertEqual(events, [.success(1)])
}
```
...

I think we are actually missing those tests for observable sequence subscribe extensions, so feel free to add them also.